libctru/include/3ds/services/fspxi.h

```diff
@@ -0,0 +1,645 @@
+
+/**
+ * @brief Sets the CARDSPI bus mode.
+ * @param busMode Bus mode to set.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(SetCardSpiBusMode, FS_CardSpiBusMode busMode);
+
+/**
+ * @brief Sends initialization info to ARM9
+ * @param unk FS sends *(0x1FF81086)
+ */
+DEFINE_PXIFS_SERVICE_METHOD(SendInitializeInfoTo9, u8 unk);
+
+/**
+ * @brief Creates ext save data.
+ * @param info Info of the save data.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(CreateExtSaveData, FS_ExtSaveDataInfo info);
+
+/**
+ * @brief Deletes ext save data.
+ * @param info Info of the save data.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(DeleteExtSaveData, FS_ExtSaveDataInfo info);
+
+/**
+ * @brief Enumerates ext save data.
+ * @param idsWritten Pointer to output the number of IDs written to.
+ * @param idsSize Size of the IDs buffer.
+ * @param mediaType Media type to enumerate over.
+ * @param idSize Size of each ID element.
+ * @param shared Whether to enumerate shared ext save data.
+ * @param ids Pointer to output IDs to.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(EnumerateExtSaveData, u32* idsWritten, u32 idsSize, FS_MediaType mediaType, u32 idSize, bool shared, u8* ids);
+
+/**
+ * @brief Gets a special content's index.
+ * @param index Pointer to output the index to.
+ * @param mediaType Media type of the special content.
+ * @param programId Program ID owning the special content.
+ * @param type Type of special content.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(GetSpecialContentIndex, u16* index, FS_MediaType mediaType, u64 programId, FS_SpecialContentType type);
+
+/**
+ * @brief Gets the legacy ROM header of a program.
+ * @param mediaType Media type of the program.
+ * @param programId ID of the program.
+ * @param header Pointer to output the legacy ROM header to. (size = 0x3B4)
+ */
+DEFINE_PXIFS_SERVICE_METHOD(GetLegacyRomHeader, FS_MediaType mediaType, u64 programId, u8* header);
+
+/**
+ * @brief Gets the legacy banner data of a program.
+ * @param mediaType Media type of the program.
+ * @param programId ID of the program.
+ * @param banner Pointer to output the legacy banner data to. (size = 0x23C0)
+ * @param unk Unknown. Always 1?
+ */
+DEFINE_PXIFS_SERVICE_METHOD(GetLegacyBannerData, FS_MediaType mediaType, u64 programId, u8* banner, u32 unk);
+
+/**
+ * Unknown command 3D
+ * @param unk Unknown. Transaction?
+ */
+DEFINE_PXIFS_SERVICE_METHOD(0x3D, u32 unk);
+
+/// Deletes the 3DS SDMC root.
+DEFINE_PXIFS_SERVICE_METHOD(DeleteSdmcRoot, void);
+
+/// Deletes all ext save data on the NAND.
+DEFINE_PXIFS_SERVICE_METHOD(DeleteAllExtSaveDataOnNand, void);
+
+/// Initializes the CTR file system.
+DEFINE_PXIFS_SERVICE_METHOD(InitializeCtrFilesystem, void);
+
+/// Creates the FS seed.
+DEFINE_PXIFS_SERVICE_METHOD(CreateSeed, void);
+
+/**
+ * @brief Gets the CTR SDMC root path.
+ * @param out Pointer to output the root path to.
+ * @param length Length of the output buffer.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(GetSdmcCtrRootPath, u8* out, u32 length);
```

Comment:
What's the actual output type of GetSdmcCtrRootPath, is it UTF-8, UTF-16 or some other raw data?

Reply:
It is UTF-16, according to 3dbrew. I assume the output should be a u16 to reflect this?

Reply:
Correct.